### DISCUSSION TITLE
Using latest GNU Make in formula build section

### DISCUSSION THREAD

**Discussion by lelegard** [Homebrew/discussions#5620] [Writing Formulae/Casks] [answered]
Hi,

In the next version of a formula (`tsduck`), GNU Make version 4+ will be required. Up to the current version of the formula, the version 3.81 of the GNU Make which is installed with macOS was sufficient. This is no longer true.

Now, when building the formula, we need the "make" which comes with Homebrew (version 4.4.1 as of today). However, that make is installed as gmake on macOS. On Linux, we don't need it.

Being ignorant in Ruby, I need some advice on the best way to update the formula so that:

1. `depends_on "make" => :build` is used on macOS only
2. In the install section, the various `system "make" ...` commands become `system "gmake"` on macOS only.

In the past, I have seen that some potentially complicated things were elegantly solved by Homebrew maintainers in one single obscure line (obscure to me). I am curious to see how to solve this one the right way.

**Comment by carlocab** [comment, marked as answer]
You could do
```ruby
on_macos do
  depends_on "make" => :build
end
```
and then in the `install` method you should be able to call `gmake` unconditionally since that will still resolve to the right executable on Linux.

I'm also tempted to say that we should make the `make` dependency unconditional for simplicity, but I'll leave that up to you.
